### PR TITLE
Creating the Conflicts section in the viewlet

### PR DESCRIPTION
**Pull request by jpricketMSFT**
These changes create the conflict section in the SCM viewlet and allow the user to see the diff of what's local and what's on the server. It does not provide a way to resolve the conflicts yet.

### CODE REVIEW

File: src/tfvc/tfvcscmprovider.ts [hunk 1]
```diff
@@ -220,19 +219,6 @@ export class TfvcSCMProvider implements SCMProvider {
         }
     }
 
-    private getTitle(resource: Resource): string {
```
Comment (jpricketMSFT) [inline src/tfvc/tfvcscmprovider.ts]: I moved this into the Resource class.

File: src/tfvc/scm/resource.ts [hunk 2]
```diff
@@ -37,10 +46,42 @@ export class Resource implements SCMResource {
      */
     public GetServerUri(): Uri {
         const serverItem: string = this._change.sourceItem ? this._change.sourceItem : this._change.serverItem;
-        const versionSpec: string = "C" + this._change.version;
+        // For conflicts set the version to "T"ip so that we will compare against the latest version
+        const versionSpec: string = this.HasStatus(Status.CONFLICT) ? "T" : "C" + this._change.version;
         return Uri.file(serverItem).with({ scheme: TfvcSCMProvider.scmScheme, query: versionSpec });
     }
 
+    public GetTitle(): string {
```
Comment (jpricketMSFT) [inline src/tfvc/scm/resource.ts]: This was moved here from the scmprovider class and then altered to handle conflicts.

File: src/tfvc/scm/model.ts [hunk 3]
```diff
@@ -109,15 +111,22 @@ export class Model {
 
     private async update(): Promise<void> {
         const changes: IPendingChange[] = await this._repository.GetStatus();
+        const foundConflicts: IConflict[] = await this._repository.FindConflicts();
+
+        //if (foundConflicts.find(c => c.type === ConflictType.NAME_AND_CONTENT || c.type === ConflictType.RENAME)) {
+        //    TODO: Send telemetry
```
Comment (jpricketMSFT) [inline src/tfvc/scm/model.ts]: I didn't see a way to get the Telemetry object from here. Perhaps we need to move it to a singleton like the TFVCOutput class.

File: src/tfvc/scm/model.ts [hunk 4]
```diff
@@ -137,10 +146,26 @@ export class Model {
...
         this._excludedGroup = new ExcludedGroup(excluded);
 
         this._onDidChange.fire(this.Resources);
     }
+
+    private conflictMatchesPendingChange(change: IPendingChange, conflict: IConflict): boolean {
+        let result: boolean = false;
+        if (change && change.localItem && conflict && conflict.localPath) {
+            // TODO: If resource or conflict are renames we have a lot more work to do
+            //       We are postponing this work for now until we have evidence that it happens a lot
```
Comment (jeffyoung) [inline src/tfvc/scm/model.ts]: Would we want to send telemetry here as well?  Or is the comment above at the location where we'd track this?

File: src/tfvc/scm/model.ts (hunk 3 above)
Comment (jeffyoung) [inline src/tfvc/scm/model.ts]: Yes, I think that's a good idea.  We should have the ability to send telemetry from wherever we need to.

File: src/tfvc/scm/model.ts (hunk 4 above)
Comment (jpricketMSFT) [inline src/tfvc/scm/model.ts]: Here we are inside a loop, so we would want it above only

**Pull request merged**